dispatch/tests/test_api_pages.py

```diff
@@ -0,0 +1,210 @@
+from django.core.urlresolvers import reverse
+
+from rest_framework import status
+
+from dispatch.tests.cases import DispatchAPITestCase
+
+from dispatch.apps.content.models import Page
+
+class PagesTest(DispatchAPITestCase):
+
+    def _create_page(self):
+        """
+        Create Dummy Page
+        """
+
+        url = reverse('api-pages-list')
+
+        data = {
+          "title": "Test Page",
+          "slug": "test-page",
+          "snippet": "This is a test snippet",
+          "content": [
+            {
+              "type": "paragraph",
+              "data": "This is some paragraph text"
+            }
+          ]
+        }
+
+        return self.client.post(url, data, format='json')
+
+    def test_create_page_unauthorized(self):
+        """
+        Create page while unauthorized should fail with unauthenticated request
+        """
+        # Clear authentication credentials
+        self.client.credentials()
+
+        response = self._create_page()
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_create_empty_page(self):
+        """
+        Create page should fail with empty payload
+        """
+
+        url = reverse('api-pages-list')
+
+        response = self.client.post(url, None, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_create_incomplete_page(self):
+        """
+        Creat Page should fail with missing required fields
+        """
+
+        url = reverse('api-pages-list')
+
+        # Page missing content, snippet, slug
+        data = {
+            "title": "Test Page"
+        }
+
+        response = self.client.post(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        count = Page.objects.all().count()
+
+        self.assertEqual(count, 0)
+
+
+    def test_create_page(self):
+        """
+        Ensure we can make a page
+        """
+
+        response = self._create_page()
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # Check Data
+        self.assertEqual(response.data['title'], 'Test Page')
+        self.assertEqual(response.data['slug'], 'test-page')
+        self.assertEqual(response.data['snippet'], "This is a test snippet")
+        self.assertEqual(response.data['content'][0]['type'], 'paragraph')
+        self.assertEqual(response.data['content'][0]['data'], 'This is some paragraph text')
+
```

Comment:
I should have mentioned this earlier, but I was thinking it would be good to add an additional assertion to check that the page was actually saved to the database. Something like this:

```py
try:
  page = Page.objects.get(pk=response.data['id'])
except Page.DoesNotExist:
  self.fail("The page should exist in the database")

self.assertEqual(page.title, 'Test page')
self.assertEqual(page.slug, 'test-page')
self.assertEqual(page.snippet, 'This is a test snippet')
self.assertEqual(page.content[0]['type'], 'paragraph')
self.assertEqual(page.content[0]['data'], 'This is some paragraph text')
```

Let's also do this for the update tests below